crates/uv-cli/src/lib.rs

```diff
@@ -2231,6 +2233,27 @@ pub struct ToolUninstallArgs {
     pub all: bool,
 }
 
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct ToolUpgradeArgs {
+    /// The name of the tool to upgrade.
+    #[arg(required = true)]
+    pub name: Option<PackageName>,
+
+    /// Upgrade all tools.
+    #[arg(long, conflicts_with("name"))]
+    pub all: bool,
+
+    #[command(flatten)]
+    pub installer: ResolverInstallerArgs,
```

Comment:
Does including this mean we support things like `--upgrade-package` here now? Is that what we want?

Reply:
Good point. The following approach works, and there are cases where people might want to upgrade tools to a specific version as well. Maybe we should support this, wdyt?

```cargo run -- tool upgrade poetry --upgrade-package poetry==1.5```